content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,170 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to the models in your schema, results objects, queries, or Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/client-extensions-model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/client-extensions-query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To create Prisma Client extensions, use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods), and the appropriate extension type (`model`, `client`, `query` or `result`).
+
+You can use [multiple independent extensions at the same time](#multiple-extensions), each in is own isolated instance of Prisma Client.
```

Comment:
Are those really isolated instances if you keep adding instances to the already modified instance?

Reply:
@janpio You mean if you choose to stack or combine instances? If so, I can add something here like "unless you deliberately stack or combine instances" and link there to the section on conflicts below.

Reply:
Yeah, very nitpicky on the words. The "isolated instance" felt a bit weird to me - can not even say why exactly.

Reply:
That's fair enough - docs should be accurate, especially highly technical ones like Prisma's

Reply:
OK, fixed!